src/TokenGenerator.php

```diff
@@ -0,0 +1,145 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Solid\Auth;
+
+use Pdsinterop\Solid\Auth\Utils\Jwks;
+use Pdsinterop\Solid\Auth\Enum\OpenId\OpenIdConnectMetadata as OidcMeta;
+
+class TokenGenerator
+{
+    ////////////////////////////// CLASS PROPERTIES \\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+    /** @var Config */
+    public $config;
+
+    //////////////////////////////// PUBLIC API \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+    final public function __construct(
+        Config $config
+    ) {
+        $this->config = $config;
+    }
+	
+	public function generateRegistrationAccessToken($clientId, $privateKey) {
+		$issuer = $this->config->getServer()->get(OidcMeta::ISSUER);
+
+		// Create JWT
+		$signer = new \Lcobucci\JWT\Signer\Rsa\Sha256();
+		$keychain = new \Lcobucci\JWT\Signer\Keychain();				
+		$builder = new \Lcobucci\JWT\Builder();
+		$token = $builder
+			->setIssuer($issuer)
+            ->permittedFor($clientId)
+			->set("sub", $clientId)
+			->sign($signer, $keychain->getPrivateKey($privateKey))
+			->getToken();
+		$result = $token->__toString();
+		return $token->__toString();
+	}
+		
+	public function generateIdToken($accessToken, $clientId, $subject, $nonce, $privateKey) {
+		$issuer = $this->config->getServer()->get(OidcMeta::ISSUER);
+
+        $jwks = $this->getJwks();
+		$tokenHash = $this->generateTokenHash($accessToken);
+
+		// Create JWT
+		$signer = new \Lcobucci\JWT\Signer\Rsa\Sha256();
+		$keychain = new \Lcobucci\JWT\Signer\Keychain();
+		$builder = new \Lcobucci\JWT\Builder();
+		$token = $builder
+			->setIssuer($issuer)
+            ->permittedFor($clientId)
+			->setIssuedAt(time())
+			->setNotBefore(time() - 1)
+			->setExpiration(time() + 14*24*60*60)
+			->set("azp", $clientId)
+			->set("sub", $subject)
+			->set("jti", $this->generateJti())
+			->set("nonce", $nonce)
+			->set("at_hash", $tokenHash) //FIXME: at_hash should only be added if the response_type is a token
+			->set("c_hash", $tokenHash) // FIXME: c_hash should only be added if the response_type is a code
+			->set("cnf", array(
+				"jwk" => $jwks['keys'][0]
+			))
+			->withHeader('kid', $jwks['keys'][0]['kid'])
+			->sign($signer, $keychain->getPrivateKey($privateKey))
+			->getToken();
+		$result = $token->__toString();
```

Comment:
Might as well just return `$token->__toString()` directly, no need for a var.

Reply:
agreed, fixed.